src/foam/nanos/pm/PM.js

```diff
@@ -82,6 +101,24 @@ foam.CLASS({
       javaCode: `
     fm.foldForState(getClassType().getId()+":"+getName(), getStartTime(), getTime());
       `
+    },
+    {
+      name: 'error',
+      args: [
+        { name: 'x', type: 'Context' },
+        { name: 'message', type: 'Object...' }
+      ],
+      javaCode: `
+        String str = "";
+        setIsError(true);
+        for (Object s: message){
+          if ( s instanceof java.lang.Exception ) {
+            str += s.toString() + ",";
```

Comment:
so here you want to setException(s), and set errorMessage to the remaining args - combined.